### ISSUE TITLE
Dashboard : Data is not plotted for non Numeric field 

### ISSUE THREAD

**Issue by ktx-akshay** [openobserve/openobserve#3815]
### Which OpenObserve functionalities are the source of the bug?

dashboards

### Is this a regression?

Yes

### Description

- 

![Screenshot 2024-06-25 160439](https://github.com/openobserve/openobserve/assets/172245377/67ba884d-856a-4127-a20f-905750fa7df2)

**Comment by gaby** [comment]
@ktx-akshay @ktx-kirtan This happens with the `Metric` visualization too. I have 4-5 labels on each prometheus metric and I can't use them because of this.

**Comment by prabhatsharma** [comment]
min(not a number) is not defined. This is right behaviour

**Comment by gaby** [comment]
@prabhatsharma ELK lets you do this since around 2016, this is done by making non-integer types as `.keyword` fields. This makes them aggregatable.

https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html

**Comment by prabhatsharma** [comment]
I think ES is different. However, irrespective, instead of trying to support it simply because ES supports it, I would ask you, what do you gain by doing this? I see no logical output from this.

`min ("hello", "world", "from", "Utopia")` mathematically does not make sense unless one tries to do jugglery around converting text to numbers and then doing operations on it. There possibly might be an obscure use case which I am not aware of but I believe that there are other ways to handle that by converting text to numbers using VRL functions.

**Comment by gaby** [comment]
@prabhatsharma Yeah that's true. I think the main use case for strings would be "Top Hit" aggregation, which is typically sorted by timestamp.

If I rum the visualization it should return the Last Value of `field` soeted by timestamp regardless of data type.

**Comment by prabhatsharma** [comment]
I would rather not do it.

**Comment by gaby** [comment]
@prabhatsharma That would limit what can be done with dashboards. Can the `_timestamp` field be used ?

**Comment by prabhatsharma** [comment]
I don't think it limits any legitimate use case. Yes, `_timestamp` can be used.
